detox/src/Detox.js

```diff
@@ -65,13 +66,41 @@ class Detox {
     if (!deviceClass) {
       throw new Error(`'${deviceConfig.type}' is not supported`);
     }
-
     const deviceDriver = new deviceClass(this.client);
-    this.device = new Device(deviceConfig, sessionConfig, deviceDriver);
+    const binaryPath = await this._getBinaryPath(deviceConfig, deviceDriver);
+
+    this.device = new Device(deviceConfig, sessionConfig, deviceDriver, binaryPath);
     await this.device.prepare(params);
     global.device = this.device;
   }
 
+  async _getAppName(deviceDriver) {
```

Comment:
I have a feeling that trying to extract the appName from project files will get us in lots of trouble. There are endless possibilities for configuring projects (especially if there are multiple targets on iOS or multiple buildFlavors on Android, and `build.gradle` might be in custom directories (not necessarily in `/app/`